README.md

```diff
@@ -645,6 +620,89 @@ const json2csvParser = new Parser({
 const csv = json2csvParser.parse(myData);
 ```
 
+#### AsyncParser
+
+Async parse have been simplified to be a class with a single `parse` method which replaces the previous `fromInput` method. `throughTransform` and `toOutput` can be replaced by `pipe` method or the newer `pipeline` utility.
+
+What used to be
+```js
+const { AsyncParser } = require('json2csv');
+const json2csvParser = new AsyncParser();
+const csv = await json2csvParser.fromInput(myData).throughTransform(myTransform).toOutput(myOutput);
+```
+
+should be replaced by
+
+```js
+const { AsyncParser } = require('json2csv');
+const json2csvParser = new AsyncParser();
+json2csvParser.parse(myData.pipe(myTransform)).pipe(myOutput);
+```
+
+The `promise` method has been kept but it doesn't take any argument as it used to. Now it always keeps the whole CSV and returns it.
+
+
+What used to be
+```js
+const { AsyncParser } = require('json2csv');
+const json2csvParser = new AsyncParser();
+const csv = await json2csvParser.fromInput(myData).promise();
+```
+
+should be replaced by
+
+```js
+const { AsyncParser } = require('json2csv');
+const json2csvParser = new AsyncParser();
+const csv = await json2csvParser.parse(myData).promise();
+```
+
+If you want to wait for the stream to finish but not keep the CSV in memory you can use the `stream.finished` utility from Node's stream module.
+
+Finally, the `input`, `transform` and `processor` properties have been remove.
+`input` is just your data stream.
+`transform` and `processor` are equivalent to the return of the `parse` method.
+
+Before you could instantiate an `AsyncParser` and push data into it. Now you can simply pass the data as the argument to the `parse` method if you have the entire dataset or you can manually create an array and push data to it.
+
+What used to be
+
+```js
+asyncParser.processor
+  .on('data', (chunk) => (csv += chunk.toString()))
+  .on('end', () => console.log(csv))
+  .on('error', (err) => console.error(err));
+
+myData.default.forEach(item => asyncParser.input.push(item));
```

Comment:
Where does `myData` come from in this example, and why does it have `default`?

Reply:
`myData` is the same myData as in all the other examples.. any array.
It doesn't have `default`. I'll remove that.